src/hlstrafe.cpp

```diff
@@ -498,6 +498,30 @@ namespace HLStrafe
 		avec[1] = wishvel[1];
 	}
 
+	void UpdateLookAtViewangle(const PlayerData& player, CurrentState& curState)
+	{
+		float target[3];
+		target[0] = curState.TargetYawLookAtOrigin[0] + curState.Parameters.Parameters.LookAt.X;
+		target[1] = curState.TargetYawLookAtOrigin[1] + curState.Parameters.Parameters.LookAt.Y;
+		target[2] = curState.TargetYawLookAtOrigin[2] + curState.Parameters.Parameters.LookAt.Z;
+
+		float difference[3];
+		VecSubtract<float, float, 3>(target, player.Origin, difference);
+
+		double yaw = 0;
+		double pitch = 0;
+		if (difference[0])
+			yaw = Atan2(difference[1], difference[0]) * M_RAD2DEG;
+
+		double z = player.Origin[2] - target[2];
+		double xy = Distance<float, float, 2>(target, player.Origin);
+		if (xy)
```

Comment:
```suggestion
		if (z)
```
i think you meant this?

Reply:
alright now i am confused, isn't `xy` denominator in this case? So it shouldn't be 0. That's what I think. Then I read the docs on https://en.cppreference.com/w/cpp/numeric/math/atan2 and they suggest that the errors *might* occur. Then the bullet points affirm the 0 error will not occur.... So no check?

Reply:
![image](https://user-images.githubusercontent.com/29143195/211125261-c8850208-0e7b-4959-b5f2-7e76b8d8eeec.png)

Reply:
Hmm okay